code/modules/reagents/metabolism.dm

```diff
@@ -207,21 +207,23 @@
 			addiction_list.Remove(R)
 			continue
 
-		addiction_list[R] += 1
-		if(!parent.chem_effects[CE_PURGER])
-
+		if(!parent.chem_effects[CE_PURGER] && ishuman(parent))
+			addiction_list[R] += 1
 			switch(addiction_list[R])
-				if(1 to 10)
+				if(1 to 20)
 					R.addiction_act_stage1(parent)
-				if(10 to 20)
-					R.addiction_act_stage2(parent)
 				if(20 to 30)
-					R.addiction_act_stage3(parent)
+					R.addiction_act_stage2(parent)
 				if(30 to 40)
+					R.addiction_act_stage3(parent)
+				if(40 to 50)
 					R.addiction_act_stage4(parent)
-				if(40 to INFINITY)
-					R.addiction_end(parent)
-					addiction_list.Remove(R)
+				if(50 to INFINITY)
+					if((parent.stats.getPerk(PERK_ALCOHOLIC) && istype(R, /datum/reagent/ethanol)) || (parent.stats.getPerk(PERK_DRUG_ADDICT) && istype(R, /datum/reagent/stim)))
+						R.addiction_act_stage4(parent)
```

Comment:
Maybe make it cycle back to the first addiction stage so it loops.

Reply:
why? it doesn't make sense